service/src/main/java/uk/gov/hmcts/reform/fpl/config/ServiceTokenGeneratorConfiguration.java

```diff
@@ -0,0 +1,26 @@
+package uk.gov.hmcts.reform.fpl.config;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cloud.openfeign.EnableFeignClients;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Lazy;
+import uk.gov.hmcts.reform.authorisation.ServiceAuthorisationApi;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGeneratorFactory;
+
+@Configuration
+@Lazy
+@EnableFeignClients(basePackageClasses = ServiceAuthorisationApi.class)
+public class ServiceTokenGeneratorConfiguration {
+
+    @Bean
+    public AuthTokenGenerator serviceAuthTokenGenerator(
+        @Value("${idam.s2s-auth.totp_secret}") final String secret,
```

Comment:
Let's remove final keyword from constructor properties. We will introduce check style rule that will forbid reassignment.